File: src/census/src/infrastructure/logging/LogBootstrapper.js
import BrowserIdEnricher from "./BrowserIdEnricher";
const StructuredLog = require('structured-log');
const SeqSink = require('structured-log-seq-sink')

class LogBootstrapper {
    constructor(configuration, version) {
        this.configuration = configuration;
        this.version = version;

        this.bootstrap = this.bootstrap.bind(this);
    }

    bootstrap() {

        var levelSwitch = new StructuredLog.DynamicLevelSwitch(this.configuration.Logging.LogEventLevel)

        var consoleSink = new StructuredLog.ConsoleSink({
            console: this.undecoratedConsole,
            includeTimestamps: true,
        });

        var seqSink = new SeqSink({
            url: this.configuration.Logging.Seq.Uri,
            apiKey: this.configuration.Logging.Seq.ApiKey,
            levelSwitch: levelSwitch,
            suppressErrors: false
        });

        var logger = StructuredLog.configure()
            .enrich({
                "ApplicationName": "Hipster Census",
                "ApplicationVersion": this.version.Version,
                "ProcessName": "browser"
            })
            .enrich(new BrowserIdEnricher().enrich)
            .writeTo(consoleSink)
            .writeTo(seqSink)
            .create();

        logger.info("Application online");

        return logger;
    }
}

export default LogBootstrapper;

File: src/Census.Api/Infrastructure/Logging/LogBootstrapper.cs
using Census.Api.AppSettings;
using Serilog;
using Serilog.Core;
using Serilog.Enrichers;
using Serilog.Events;

namespace Census.Api.Infrastructure.Logging
{
    public static class LogBootstrapper
    {
        public static void Bootstrap(ApplicationSettings applicationSettings, LoggingSettings loggingSettings)
        {
            var controlLevelSwitch = new LoggingLevelSwitch(LogEventLevel.Verbose);

            Log.Logger = new LoggerConfiguration()
                         .MinimumLevel.ControlledBy(controlLevelSwitch)
                         .Enrich.FromLogContext()
                         .Enrich.WithProperty("ApplicationName", applicationSettings.Name)
                         .Enrich.WithProperty("ApplicationVersion", typeof(LogBootstrapper).Assembly.GetName().Version)
                         .Enrich.With<FrameworkSpamReductionEnricher>()
                         .Enrich.With<MachineNameEnricher>()
                         .Enrich.With<EnvironmentUserNameEnricher>()
                         .Enrich.With<ThreadIdEnricher>()
                         .Enrich.With<ProcessIdEnricher>()
                         .Enrich.With<ProcessNameEnricher>()
                         .Enrich.With<ExceptionDataEnricher>()
                         .WriteTo.Seq(loggingSettings.Seq.Uri.ToString(),
                                      apiKey: loggingSettings.Seq.ApiKey,
                                      controlLevelSwitch: controlLevelSwitch)
                         .CreateLogger();

            Log.Information("Application online");
        }
    }
}